js/src/data.ts

```diff
@@ -177,16 +180,16 @@ export class Data<T extends DataType = DataType> {
                 // if we have a nullBitmap, truncate + slice and set it over the pre-filled 1s
                 if (this.nullCount > 0) {
                     nullBitmap.set(truncateBitmap(offset, length, this.nullBitmap), 0);
+                    Object.assign(this, { nullBitmap });
+                } else {
+                    Object.assign(this, { nullBitmap, _nullCount: 0 });
```

Comment:
Here we can set the null count to 0 since we just created a new nullBitmap that is all valid values.